### ISSUE TITLE
False positive for pseudos and Less guard in selector-descendant-combinator-no-non-space

### ISSUE THREAD

**Issue by thybzi** [stylelint/stylelint#2110]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Using CSS pseudo selector (`:hover`, `:not()`, `::after` etc.) and LESS guard (`when`) in the same line leads to `selector-descendant-combinator-no-non-space` false positive.

> Which rule, if any, is this issue related to?

`selector-descendant-combinator-no-non-space`

> What CSS is needed to reproduce this issue?

```less
:hover when (@a = 42) {
    color: red;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
    "rules": {
        "selector-descendant-combinator-no-non-space": true
    }
}
```

> Which version of stylelint are you using?

e.g. `7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint test.less`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to LESS `when` syntax: http://lesscss.org/features/#css-guards-feature

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.less
 1:12  ×  Unexpected " (@a = 42)"   selector-descendant-combinator-no-non-space
```

**Comment by thybzi** [comment]
Concerns not only `:not()`, but all pseudo selectors. Updated the issue description.

**Comment by evilebottnawi** [comment]
@thybzi Thanks for feedback, today i can see this.

**Comment by evilebottnawi** [comment]
@thybzi can you provide full example, i can't find using `pseudo` with `when` in docs?

**Comment by thybzi** [comment]
Hi! I have provided the example in the head of this issue:
```less
:hover when (@a = 42) {
    color: red;
}
```

**Comment by evilebottnawi** [comment]
@thybzi I am on the weekends :smile: Tomorrow it'll :+1:

**Comment by thybzi** [comment]
Okay, waiting for this. Thank you!

**Comment by evilebottnawi** [comment]
@thybzi sorry, very busy last month :sob: i can try resolve himself if you need this immediately :smile:

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2557
